sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -583,7 +583,9 @@ object FunctionRegistry {
           val validParametersCount = constructors
             .filter(_.getParameterTypes.forall(_ == classOf[Expression]))
             .map(_.getParameterCount).distinct.sorted
-          val expectedNumberOfParameters = if (validParametersCount.length == 1) {
+          val expectedNumberOfParameters = if (validParametersCount.length == 0) {
+            throw new AnalysisException(s"$name is an invalid function.")
+          } else if (validParametersCount.length == 1) {
             validParametersCount.head.toString
           } else {
             validParametersCount.init.mkString("one of ", ", ", " and ") +
```

Comment:
can we just handle empty `validParametersCount` here? I think it's better to have a unified error message:
```
Invalid number of arguments for function $name. Expected .....
```